### PR TITLE
[Ready] Mining Loot Rework

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

This PR reworks, rebalances and/or replaces various mining loot.
Included changes to (might be forgetting some):
Soulstone, Lava Staff, Staff of Storms, Eye of God, Cursed Katana, Antique Laser, Anomalous Crystal, Memento Mori, Berserker Hardsuit, H.E.C.K. Suit, Spellblade, Dusty Shard, Voice of God
See more information here: https://hackmd.io/PrTwt3qqTAyYJK6U_aTNSw?view (Written a while ago, might be outdated)
Cleans up some code on the loot and megafauna
Splits the necropolis_chests.dm file
Removes the unused old colossus loot, the blackbox

## Why It's Good For The Game

It makes the loot less broken, more fun and shouldn't have as many one click wins.
It isn't lame like beestations overhaul you get cool zappy thunderbolts fuck yea

## Changelog
:cl: Fikou, MeyHaZah, some sprites from /vg/station
balance: mining soulstone can no longer grab sleeping/unconscious people
balance: voice of god can immobilize in addition to knocking down
balance: anomalous crystal now has a time for it to be used, its snow theme is no longer lethal, you can run from the clown one
balance: lava staff is weaker and its lava is weaker
balance: wand of fireball removed from ash drake loot
balance: drake armor nerfed values, especially to bullet armor
balance: berserker suit has more bullet armor, charge gain doubled
balance: hecksuit is spaceproof, you can use it to gib corpses and heal from them
balance: eye of god loses its xray, you can now use it to scan targets to lower their attack speed
balance: cursed katana is now an extendable organ, lower damage, attacks fauna faster, has a combo system
balance: staff of storms no longer makes storms, it can  create thunderbolts, the thunderbolts are boosted in weather conditions
balance: winter biodome is a bit harder, its antique laser has now been replaced with a freeze cube in a vault
balance: spellblade from bubblegum is replaced with a soulscythe
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
NOOO CONFLICTS

**Comment by tralezab** [comment]
you should describe what you actually did to each one

**Comment by ChungusGamer666** [comment]
i literally don't know a single thing this PR is doing

**Comment by Fikou** [comment]
> you should describe what you actually did to each one

done

**Comment by tralezab** [comment]
wow your "doc" says nothing

can you just put your damn changes in the OP like everyone else jesus

**Comment by Fikou** [comment]
> wow your "doc" says nothing
> 
> can you just put your damn changes in the OP like everyone else jesus

its not a doc and it lists all the changes

**Comment by tralezab** [comment]
"An unique scythe that can be possessed by spirits to interact with the world"

This blows

**Comment by Fikou** [comment]
> "An unique scythe that can be possessed by spirits to interact with the world"
> 
> This blows

yes this is one of the things i havent finished yet :)

**Comment by AMonkeyThatCodes** [comment]
arm is starting to sound like me

**Comment by tralezab** [comment]
hm? this is, after all, my first post in this thread. love your pr fik ok gotta work on engravings see ya around. crack eggs in your ramen

**Comment by ChungusGamer666** [comment]
Too unfinished to make a solid opinion on it for now. I have mixed feelings about the hackmd.

**Comment by ATH1909** [comment]
> Soulstone - Just make it the variant that can’t grab sleeping/unconscious people. - The only way this is relevant for a miner is a oneshot combo with Voice of God sleep command.

Didn't https://github.com/tgstation/tgstation/pull/60002 patch this combo? This change isn't necessary and could confuse cultists who might think that their soulstones work like the miner versions, IMO.

> Voice of God - Has an immobilize command in addition to the knockdown. - Spices it up and you can choose between immobilizing while they can still attack you or knocking them down while they can still move.

Very good and interesting addition.

> Lava Staff - Make it create “colder” lava that deals less damage. - Something that creates infinite one-shotting traps is kinda stupid.

Fair enough.

> Staff of Storms - No longer creates storms, by using it in hand you can stop all kinds of weather, but if you aim it at a tile, a telegraphed thunder will appear, dealing a bunch of damage. This thunderbolt will have a giant boost if in an area with ongoing weather effects. - This one isn’t bad, but I feel like weather effects are kind of boring and the station doesn’t really have any counterplay other than to wait it out.

Seems like that'd just turn it into a lame lava staff knockoff. I'd instead go with making the storm go away if the miner drops or sheathes their staff of storms, so you can stop the storm by just dunking on or disarming the person who's causing it. Maybe add a glow or something around the staff of storms user so that people can easily identify who they need to lynch.

> Eye of God - No more X-Ray, it can keep the mesons/night vision. They give you a scan ability, using it lets you target a mob with your cursor. By clicking on a mob, that mob becomes “staggered”. You now have an internal pinpointer to it, it gains and outline and it doubles the cooldown on their ranged attacks (or every attack on megafauna) for 30 seconds. - This one is really boring, it’s just night vision X-Ray you can’t take off. It also becomes stupid with hierophant club.

Sad. I'd personally balance this by giving it negative flash protection, but I guess this effect is okay enough. Please add an option to toggle the meson vision if you haven't already. Can you (and other people) see the outlined person through walls, like what the Glowing effect from Minecraft does?

> Cursed Katana - Is no longer a katana subtype. It’s a sort of organ/implant you can extend/retract, deals 15 damage and you can hack and slash at stuff with LMB/RMB, with certain combinations of inputs making special attacks. If the target is fauna, every slash has a low attack cooldown. The curse is mauling your hand if you retract the katana without using it on an alive target. - Another kind of boring one, and way too station centered. You lose your hand slot making mining way harder and instead get a super insane melee weapon with 40 damage and 50% block chance you can’t drop.

The idea of giving the cursed katana a force equal to that of a fucking roundstart circular saw disgusts me. The low attack cooldown and combos are kind of interesting... but would give this item a LOT of overlap with the design of the cleaving saw that the Blood Drunk Miner drops. I'd just drop the force to, say, 30 and call it a day, tbh. Maybe add an action button that you can press to drop the blade (and become uncursed) after you've dealt enough damage with it.

> Antique Laser - Replaced with a freeze cube, since it spawns in a winter biodome. You throw it to freeze something. - You shouldn’t get a powerful traitor objective from a ruin.

I actually quite like the idea of having items like the antique laser and black box be available in ruins; it gives antags who need to steal those items a(n unlikely) way out if someone else has already stolen them and/or fucked off with them and can cause interesting misunderstandings between sec and miners. A similar thing also applies to photos of the blueprints and making your own reactive tele armor, although those are much more consistent. I'd maybe even go as far as to give this ruin its own special signal so that traitors can know it exists before setting off on a quest to find it.

> Anomalous Crystal - Telegraph it, make snow theme not shit. - Instantly turning into a clown isn’t fun.

It's already a giant blue crystal, I dunno how much more you can telegraph it. Maybe give it some sparkles around it or something? Instantly turning into a clown if you touch (or say a phrase with "nk" in it near) the giant, sparkling, obviously magical crystal sounds fun to me.

> Memento Mori - Traps a guardian spirit/holoparasite inside it if the master reaches -100% health. Frees when out of critical. - Being unkillable with combat ability is BS.

Good change.

> Berserker Hardsuit - Buff the ratio between damage dealt and charge gained. You currently need 400 damage (or 266 from projectiles) to have it activate if I’m reading my code right. - Too weak.

Good change.

> H.E.C.K. Suit - One piece hardsuit, space-proof, lets you butcher corpses to recover, scaling with their max health. - Kind of just outclassed by drake armor, this should be an appropriate reward for Bubblegum

Sounds okay to me.

> Wand of Fireball - ???

Maybe replace it with a potion that gives the person who drinks it the firebreath mutation, ideally with said firebreath mutation being able to chromosome'd and copied by a genetics console?

> Drake Armor - ???

Drake armor is fine as it is, IMO.

> Spellblade - Replaced with: Soulscythe. An unique scythe that can be possessed by spirits to interact with the world. - It’s a gun that delimbs and recharges. It’s incredibly unfun to play against and is another wizard weapon.

This sounds like it'd have a lot of overlap with soulstones, which can turn legion corpses into ghost-controlled shades.

> Dusty Shard - Now gets 3 special mining related guardians and nothing else. These can help you mine and have special abilities for fauna fights. - The random guardians you can get are in no way helpful to mine other than the standard or support one. They are killing machines designed for the station.

Yuck. Just give the existing guardian types the ability to mine rocks with melee attacks if they can't already and call it a day. Who the fuck wants to be a simplemob whose only purpose is to mine ore, let alone one that is bound to stay near someone who will eventually leave lavaland? There's a reason why you don't see very many sentient minebots or goldgrubs, y'know.

**Comment by CRITAWAKETS** [comment]
I expected the changes to be kinda _stinky_ but i agree with all of them. For drake armor i'd suggest massively lowering the armors for anything that isnt melee, as right now drake armor is way too good onstation when it feels more like its meant to be the ultimate lavaland armor.

**Comment by tralezab** [comment]
>Who the fuck wants to be a simplemob whose only purpose is to mine ore

Our ONE minebot player is VERY angry at you rn

**Comment by Fikou** [comment]
> Didn't #60002 patch this combo? This change isn't necessary and could confuse cultists who might think that their soulstones work like the miner versions, IMO.

i started work on this pr before that, i think its still a change that isnt bad for the game, cultist soulstones get different messages from miner ones when you try to shard someone

> Seems like that'd just turn it into a lame lava staff knockoff. I'd instead go with making the storm go away if the miner drops or sheathes their staff of storms, so you can stop the storm by just dunking on or disarming the person who's causing it. Maybe add a glow or something around the staff of storms user so that people can easily identify who they need to lynch.

i think its way cooler than the lava staff :)

> Sad. I'd personally balance this by giving it negative flash protection, but I guess this effect is okay enough. Please add an option to toggle the meson vision if you haven't already. Can you (and other people) see the outlined person through walls, like what the Glowing effect from Minecraft does?

this was my plan but seeing only specific people through walls is impossible in byond without creating our own sight system

> The idea of giving the cursed katana a force equal to that of a fucking roundstart circular saw disgusts me. The low attack cooldown and combos are kind of interesting... but would give this item a LOT of overlap with the design of the cleaving saw that the Blood Drunk Miner drops. I'd just drop the force to, say, 30 and call it a day, tbh. Maybe add an action button that you can press to drop the blade (and become uncursed) after you've dealt enough damage with it.

i think you have to see it ingame to form an opinion on it

> I actually quite like the idea of having items like the antique laser and black box be available in ruins; it gives antags who need to steal those items a(n unlikely) way out if someone else has already stolen them and/or fucked off with them and can cause interesting misunderstandings between sec and miners. A similar thing also applies to photos of the blueprints and making your own reactive tele armor, although those are much more consistent. I'd maybe even go as far as to give this ruin its own special signal so that traitors can know it exists before setting off on a quest to find it.

i disagree, an easy copout that can be stored in your backpack, that has absolutely no relation to anything in a weak snowman ruin is dumb

> It's already a giant blue crystal, I dunno how much more you can telegraph it. Maybe give it some sparkles around it or something? Instantly turning into a clown if you touch (or say a phrase with "nk" in it near) the giant, sparkling, obviously magical crystal sounds fun to me.

it has a variable time so it isnt instant, there a visual effects and a balloon alert so you can run away to get away from the effects, you will still get caught if you arent aware

> This sounds like it'd have a lot of overlap with soulstones, which can turn legion corpses into ghost-controlled shades.

shades are really boring they have no abilities other than ventcrawling

> Yuck. Just give the existing guardian types the ability to mine rocks with melee attacks if they can't already and call it a day. Who the fuck wants to be a simplemob whose only purpose is to mine ore, let alone one that is bound to stay near someone who will eventually leave lavaland? There's a reason why you don't see very many sentient minebots or goldgrubs, y'know.

being a regular guardian is even worse. if your master isnt an antag and you arent a support type, you will literally sit in his body for 90% of the shift because holoparas are horrible on lavaland. mining ore is not the only purpose of these, but they will be good at that.

**Comment by AndroidMK2** [comment]
I would buff berserker's basic stats more, as I remember it is literally weaker than a regular mining suit or the bone armor which is weird couse it's name even indicates it should be a bit better, security armor might i think has higher stats than this armor.

**Comment by D4C-420** [comment]
> I would buff berserker's basic stats more, as I remember it is literally weaker than a regular mining suit or the bone armor which is weird couse it's name even indicates it should be a bit better, security armor i think has higher stats than this armor too.

When not in rage the berserkers suit has 30 melee (or non upgraded explorer suit armor). When in rage that goes up to 50 (which is 10 less then the 60 you get with fully upgraded mining amour). If I remember correctly it was supposed to be a side grade to basic armor but with melee stats like that its outclassed by every other amour mining can get. The only time its an advantage over any of mining's other armor is the anti magic that comes with it which makes zero difference on lava land. 

On a side note I few more changes I would make are icebox specific. As it stands its far too easy to just go up and grab the anti magic spear from the clockwork ruin on icebox. I would change it to be a drop from the "megafauna" ( and i use that term loosely here as the common icewelp is more dangerous then it but that is what happens when it spawns station level) so there is a bit more challenge in getting it. 

I would also make it harder to get ashdrake on icebox as currently killing 5 welps is all you need and from there you can just farm it (Mabye lower the drop rate from 100% to 50 or somthing. mabye make icewelps harder to kill. Maybe change out the armor for one with lower stats but giving immunity to icewelp breath.)

My last suggestion is for the fireball wand. Make it a flame thrower which does DOT to simple mobs. Sort of like your stealing the ashdrakes lungs (of course to people this would basically be a rechargeable flame pellet shotgun but honestly I think if it just sets people on fire its comparable enough to other things found on station).

**Comment by Serxule** [comment]
I agree with everything except for wand of fireball and god eyes, firebreath mutation is extremely lame and usually not helpful in any way other than to scare people off, especially with how easy it is to get normally, and god eyes I find have more use with xray than they do with night-vision or meson (or maybe one god eye just has meson and night-vis, but with 2 god eyes you get all 3)

**Comment by bastardblasted** [comment]
i'd say give the katana a bit more damage, i really like the combo idea but it's just straight up not worth using over pka or crusher if its only gonna do 15 damage.

**Comment by Fikou** [comment]
> When not in rage the berserkers suit has 30 melee (or non upgraded explorer suit armor). When in rage that goes up to 50 (which is 10 less then the 60 you get with fully upgraded mining amour).

your innate armor rises to 50, giving you 80 total armor

> On a side note I few more changes I would make are icebox specific. As it stands its far too easy to just go up and grab the anti magic spear from the clockwork ruin on icebox. I would change it to be a drop from the "megafauna" ( and i use that term loosely here as the common icewelp is more dangerous then it but that is what happens when it spawns station level) so there is a bit more challenge in getting it.

i havent played ice box mining for more than 2 rounds and i dont want to change it without experience with it

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Document EVERY CHANGE in the changelog. It doesn't matter if it is long, it needs to be precise.

**Comment by ANAME593** [comment]
> being a regular guardian is even worse. if your master isnt an antag and you arent a support type, you will literally sit in his body for 90% of the shift because holoparas are horrible on lavaland. mining ore is not the only purpose of these, but they will be good at that. 

Well Balancing mining loot shouldn't make it only useful for mining and equal to a item literally anybody may obtain when on station. I say make the guardians not insanely powerful on station but give them good pvp utility as well as good mining & PVE utility. thus its useful on lavaland and the station but not overly powerful. Plus a dude with a  gun can actually beat you if you have a power miner a shotgun can also dunk on you .357s aswell as most ballistic weaponary so they're not as powerful as you think. I just hope that the guardians will still be viable in PvP not sure on your changes tho other then Unga bunga useless on station

**Comment by ANAME593** [comment]
> > When not in rage the berserkers suit has 30 melee (or non upgraded explorer suit armor). When in rage that goes up to 50 (which is 10 less then the 60 you get with fully upgraded mining amour).
> 
> your innate armor rises to 50, giving you 80 total armor
> 
> > On a side note I few more changes I would make are icebox specific. As it stands its far too easy to just go up and grab the anti magic spear from the clockwork ruin on icebox. I would change it to be a drop from the "megafauna" ( and i use that term loosely here as the common icewelp is more dangerous then it but that is what happens when it spawns station level) so there is a bit more challenge in getting it.
> 
> i havent played ice box mining for more than 2 rounds and i dont want to change it without experience with it
> 
> > i'd say give the katana a bit more damage, i really like the combo idea but it's just straight up not worth using over pka/crusher or even the saw cleaver if its only gonna do 15 damage
> 
> it is way more portable than the crusher, cleaver and pka, it can block some attacks and the 15 damage stacks up pretty quick with the fast attacks and combos. yeah but that doesn't really matter plus this is tendril loot cleaver and crusher is  easier to get i think if the combos work on Players thats fine and if its stronger against lavaland fauna its a discount cleaving saw and perfect if there isn't a  resonant signal

**Comment by Fikou** [comment]
Weww bawancing mining woot shouwdn't mean make it onwy usefuw fow mining and equaw to a item witewawwy anybody may obtain when on station. I say make the guawdians not insanewy powewfuw on station but give them good pvp utiwity as weww as good mining & pve utiwity. thus its usefuw on wavawand and the station but not ovewwy powewfuw. Pwus a dude with a gun can actuawwy beat you if you have a powew minew a shotgun can awso dunk on you.357s asweww as most bawwistic weaponawy so they'we not as powewfuw as you think. I just hope that the guawdians wiww stiww be viabwe in pvp not suwe on youw changes tho othew then unga bunga usewess on station. if somehow the guawdians can be usefuw fow "pve combat" then it'd have to be insanewy usefuw in a megafauna fight as no hewp is wequiwed fow the basic fauna and i don't feew wike giving them pve combat abiwites is gonna make a fight wike ash dwake bubbwegum ow cowossus easiew ow any othew megafauna
yeah but that doesn't weawwy mattew pwus this is tendwiw woot cweavew and cwushew is easiew to get i think if the combos wowk on pwayews thats fine and if its stwongew against wavawand fauna its a discount cweaving saw and pewfect if thewe isn't a wesonant signaw

**Comment by ANAME593** [comment]
all i can say is LOL

**Comment by bastardblasted** [comment]
if you combine 2 eyes of god they should give the xray back

**Comment by D4C-420** [comment]
Move X-ray to the wisp item. As it stands it’s always been a worse eye of god in that it make it impossible to disguise yourself as someone else and it’s clear to anyone around you that you have it. I think giving it xray would be a good change the item needs to be not just a worse eye of god. 

Also I don’t like the nerf in ash drake’s melee stat from 70 to 65. Melee is the one thing other than fire that should be high on ash drake.

**Comment by ANAME593** [comment]
> Also I don’t like the nerf in ash drake’s melee stat from 70 to 65. Melee is the one thing other than fire that should be high on ash drake. Well 65 is only +5  from max explorer suit but u can use sec jumpsuits for +10 however drake armour still has ash storm protection and fire protection

**Comment by Fikou** [comment]
> if you combine 2 eyes of god they should give the xray back
> Move X-ray to the wisp item. As it stands it’s always been a worse eye of god in that it make it impossible to disguise yourself as someone else and it’s clear to anyone around you that you have it.

i have a vision and my vision does not include xray as mining loot so you can get free all access and snipe antags. if you want xray feel free to interact with another department (robotics)

> I think giving it xray would be a good change the item needs to be not just a worse eye of god.

well eye of god now works completely different 

> Also I don’t like the nerf in ash drake’s melee stat from 70 to 65. Melee is the one thing other than fire that should be high on ash drake.

its still a direct upgrade to a fully upgraded mining suit and has ash storm + fire

**Comment by D4C-420** [comment]
> i have a vision and my vision does not include xray as mining loot so you can get free all access and snipe antags. if you want xray feel free to interact with another department (robotics)
well eye of god now works completely different
its still a direct upgrade to a fully upgraded mining suit and has ash storm + fire


Fair enough.

File: code/modules/mining/lavaland/megafauna_loot.dm [hunk 1]
```diff
@@ -0,0 +1,1011 @@
...
+	icon_state = "soulslash"
+	flag = MELEE //jokair
+	damage = 15
+	light_range = 1
+	light_power = 1
+	light_color = LIGHT_COLOR_BLOOD_MAGIC
+
+/obj/projectile/soulscythe/on_hit(atom/target, blocked = FALSE)
+	if(ishostile(target))
+		damage *= 2
```
Comment (Kylerace) [inline code/modules/mining/lavaland/megafauna_loot.dm:599]: making this review so athath can put the eyes react on it

File: code/modules/antagonists/wizard/equipment/soulstone.dm [hunk 2]
```diff
@@ -53,8 +59,8 @@
 
 /obj/item/soulstone/examine(mob/user)
 	. = ..()
-	if(role_check(user) || isobserver(user))
-		if (old_shard)
+	if(role_check(user)||isobserver(user))
```
Comment (Kylerace) [inline code/modules/antagonists/wizard/equipment/soulstone.dm:62]:
```suggestion
	if(role_check(user) || isobserver(user))
```

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/mining/lavaland/megafauna_loot.dm [hunk 3]
```diff
@@ -0,0 +1,1011 @@
...
+	UnregisterSignal(user, COMSIG_HUMAN_EARLY_UNARMED_ATTACK)
+	var/datum/component/butchering/butchering = GetComponent(/datum/component/butchering)
+	butchering.butchering_enabled = FALSE
+	to_chat(user, span_notice("You lose your bloodlust."))
+
+/obj/item/clothing/head/helmet/space/hardsuit/hostile_environment/proc/butcher_target(mob/user, atom/target, proximity)
+	SIGNAL_HANDLER
+	if(!isliving(target))
+		return
+	return SEND_SIGNAL(src, COMSIG_ITEM_ATTACK, target, user)
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/modules/mining/lavaland/megafauna_loot.dm:317]: this is the same behaviour as butchering gloves, it attacks the target starting the butchering reaction

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 4]
```diff
@@ -61,6 +61,8 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 	var/tech_fluff_string = "<span class='holoparasite'>BOOT SEQUENCE COMPLETE. ERROR MODULE LOADED. THIS SHOULDN'T HAPPEN. Submit a bug report!</span>"
 	var/carp_fluff_string = "<span class='holoparasite'>CARP CARP CARP SOME SORT OF HORRIFIC BUG BLAME THE CODERS CARP CARP CARP</span>"
 	var/miner_fluff_string = "<span class='holoparasite'>You encounter... Mythril, it shouldn't exist... Submit a bug report!</span>"
+	///Are we forced to not be able to manifest/recall?
+	var/locked = FALSE
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm:65]: its changed on the memento mori obj

File: code/modules/mining/lavaland/megafauna_loot.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/mining/lavaland/megafauna_loot.dm:599]: eyes

**Comment by ExcessiveUseOfCobblestone** [comment]
pov u forget to press send on ur review

File: code/modules/mining/lavaland/tendril_loot.dm [hunk 5]
```diff
@@ -0,0 +1,1070 @@
...
+		var/list/combo_specifics = combo_list[combo]
+		if(compare_list(input_list,combo_specifics[COMBO_STEPS]))
+			INVOKE_ASYNC(src, combo_specifics[COMBO_PROC], target, user)
+			return TRUE
+	return FALSE
+
+/obj/item/cursed_katana/proc/reset_inputs(mob/user, deltimer)
+	input_list.Cut()
+	if(user)
+		balloon_alert(user, "you return to neutral stance")
```
Comment (Xander3359) [inline code/modules/mining/lavaland/tendril_loot.dm:947]: Why is this balloon alert so long WTF

File: code/modules/mining/lavaland/tendril_loot.dm [hunk 6]
```diff
@@ -0,0 +1,1070 @@
...
+
+/obj/item/cursed_katana/attack_self(mob/user)
+	. = ..()
+	reset_inputs(user, TRUE)
+
+/obj/item/cursed_katana/attack(mob/living/target, mob/user, click_parameters)
+	if(target.stat == DEAD || target == user)
+		return ..()
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
+		balloon_alert(user, "you don't want to harm!")
```
[the remark this answers is not in the document]
Comment (Mothblocks) [reply code/modules/mining/lavaland/tendril_loot.dm:912]: u can make a pr to fix it

**Pull request merged**